llvm/lib/Transforms/Coroutines/CoroSplit.cpp

```diff
@@ -726,11 +726,14 @@ static void replaceSwiftErrorOps(Function &F, coro::Shape &Shape,
 
 /// Returns all DbgVariableIntrinsic in F.
 static SmallVector<DbgVariableIntrinsic *, 8>
-collectDbgVariableIntrinsics(Function &F) {
+collectDbgVariableIntrinsics(Function &F, SmallVector<DPValue *> &DPValues) {
```

Comment:
It feels wrong to have an in parameter and a return type that are virtually the same thing.
I believe we should express this at the API more clearly by returning a pair of vectors, or make two separate functions

Reply:
SGTM. I think it'd be cleaner to have two separate functions, but that'd make it easier to accidentally ignore DPValues, so I've gone with the slightly more cumbersome returned pair.